portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -76,28 +76,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("sentimentScore", sentimentScore);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
 
     if (blobKeyString != null) {
       BlobKey blobKey = new BlobKey(blobKeyString);
       byte[] blobBytes = getBlobBytes(blobKey);
       List<EntityAnnotation> imageLabels = getImageLabels(blobBytes);
 
-      ArrayList<EmbeddedEntity> labelsEntityList = new ArrayList<EmbeddedEntity>();
-      for (EntityAnnotation label : imageLabels) {
-        EmbeddedEntity labelEntity = new EmbeddedEntity();
-        labelEntity.setProperty("description", label.getDescription());
-        labelEntity.setProperty("score", label.getScore());
-        labelsEntityList.add(labelEntity);
+      if (imageLabels == null) {
```

Comment:
My understanding is if the image isn't valide then imageLabels==null?

If so you should explain that logic as a comment.

Reply:
Yeah, that's the logic. I'll add the comment.